crates/net/network/src/transactions/fetcher.rs

```diff
@@ -78,11 +78,13 @@ pub struct TransactionFetcher<N: NetworkPrimitives = EthNetworkPrimitives> {
     /// be fetched.
     #[pin]
     pub inflight_requests: FuturesUnordered<GetPooledTxRequestFut<N::PooledTransaction>>,
-    /// Hashes that are awaiting an idle fallback peer so they can be fetched.
+    /// Hashes that are awaiting an idle fallback peer so they can be fetched grouped by peer id.
     ///
     /// This is a subset of all hashes in the fetcher, and is disjoint from the set of hashes for
     /// which a [`GetPooledTransactions`] request is inflight.
-    pub hashes_pending_fetch: LruCache<TxHash>,
+    pub hashes_pending_fetch_by_peer: LruMap<PeerId, LruCache<TxHash>, ByLength>,
```

Comment:
we can likely optimize this because we know we only have a limited amount of peers and we can clean up after a peer's session is closed, so the outer map can be a regular hashmap

Reply:
> we can clean up after a peer's session is closed

yes, it's the optimization I want to make in the future :)

Reply:
remember this was mentioned in sigp audit however, so at least we should have a metric for the size of this map to catch a memory leak early. could be that the code which removes from this map gets buggy down the road.